packages/react-charting/src/utilities/utilities.ts

```diff
@@ -747,7 +749,7 @@ export function createYAxisLabels(
         .attr('text-align', 'start')
         .append('tspan')
         .attr('id', 'LessLength')
-        .attr('x', isRtl ? xValue : x)
```

Comment:
why remove xValue?

Reply:
In cartesianChart file we are updating the state `startFromX` to the max length of the yaxis label.
``` 
if (
      this.props.chartType === ChartTypes.HorizontalBarChartWithAxis &&
      this.props.showYAxisLables &&
      this.yAxisElement
    ) {
      const maxYAxisLabelLength = calculateLongestLabelWidth(
        this.props.points.map((point: IHorizontalBarChartWithAxisDataPoint) => point.y),
        `.${this._classNames.yAxis} text`,
      );
      if (this.state.startFromX !== maxYAxisLabelLength) {
        this.setState({
          startFromX: maxYAxisLabelLength,
        });
      }
    } 
```

so in RTL mode it is adding a space infront of the label.
<img width="677" alt="Screenshot 2023-12-11 at 11 03 04 PM" src="https://github.com/microsoft/fluentui/assets/33349461/969c5b12-23e1-4cd3-a4ee-b326809990ff">

and that is why I think `xValue` should be removed from the `createYAxisLabels` function.
please correct me If I'm wrong.

Reply:
Can you test by adding a very long label like "String Very long First label" and a very short label like "Short" and share the behavior.
Basically test with more varied test cases after your fix and see if it works properly.
@ankityadav4 do you know the use of xValue property in rtl mode.

Reply:
1. Here is the example of long text label if we keep `xValue` 
<img width="500" alt="Screenshot 2023-12-13 at 9 43 16 PM" src="https://github.com/microsoft/fluentui/assets/33349461/6af5c9b9-1a74-4cfc-b86f-ae3744c62de9">

2. here is the example of long text label if we remove `xValue`
<img width="500" alt="Screenshot 2023-12-13 at 9 42 27 PM" src="https://github.com/microsoft/fluentui/assets/33349461/d04d7c21-1861-4257-a839-17cdd4411c00">

Reply:
I think we should change the approach of label width instead of taking the max length of label we can specify maxWidth and we can break text.
max 120 characters can be added for y label we can reduce that too.
<img width="500" alt="image" src="https://github.com/microsoft/fluentui/assets/33349461/7fc3090e-e7b1-4ab1-9df0-35b12fc6cfdd">

Reply:
We are working to define the guidance to handle really long and disparate labels. For now we will leave it as it is.
Also there is a property called wrapLabels in other charts to wrap for such scenarios.

Reply:
> Can you test by adding a very long label like "String Very long First label" and a very short label like "Short" and share the behavior. Basically test with more varied test cases after your fix and see if it works properly. @ankityadav4 do you know the use of xValue property in rtl mode.

This is not  needed for ticks in RTL mode. Fix seems to be fine.

Reply:
@harsh9975 We should remove the isRTL check and keep the line instead for better readability

Reply:
> @harsh9975 We should remove the isRTL check and keep the line instead for better readability

Did not get what you meant.

Reply:
I am suggesting to keep this line '.attr('x', isRtl ? xValue : x)', just removing the isRTL check here. so make it .attr('x',  x)

Reply:
but he is using x=0 for RTL. How will that work

Reply:
I am saying to change the condition in line 726 and keep this line as is.

Reply:
if I am removing the ` isRtl ? 0 : x` and keeping it directly `.attr('x', x)` there is a gap between label and graph in rtlMode thatswhy I used x=0.
<img width="500" alt="Screenshot 2023-12-18 at 5 35 04 PM" src="https://github.com/microsoft/fluentui/assets/33349461/34b0338a-b554-4d8f-ba3e-61bea5b4805b">

Reply:
Current condition looks fine.